app/models/post.rb

```diff
@@ -1,5 +1,18 @@
 class Post < ApplicationRecord
-  belongs_to :user
+  belongs_to :author, class_name: 'User'
   has_many :likes, foreign_key: :post_id
   has_many :comments, foreign_key: :post_id
+
+  before_save :update_post_counter
+  # scope :get_5_comments, ->(post) { post.comments.order('created_at DESC').limit(5) }
+
+  def update_post_counter
+    op = User.find_by_id(author.id)
+    op.posts_counter = op.posts_counter.to_i + 1
+    op.save
+  end
```

Comment:
```suggestion
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

- [ ] It would be better to  use the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private 👍🏿  


- [ ] Kindly remove all commented code from your project, you can leave comments about your code but not commented code, this will make your codebase look neat and readable 😉